src/Quartz.Tests.Unit/Simpl/JsonObjectSerializerTest.cs

```diff
@@ -1,232 +1,644 @@
+            writer.WritePropertyName("CustomProperty");
+            writer.WriteValue(calendar.CustomProperty);
+        }
 
-            CompareSerialization(trigger, systemTextJsonOnly: true);
+        protected override void DeserializeFields(JsonSerializationTestCalendar calendar, JObject source)
+        {
+            calendar.CustomProperty = source["CustomProperty"]!.Value<int>()!;
         }
 
-        private void CompareSerialization<T>(T original, Action<T, T> asserter = null, bool systemTextJsonOnly = false) where T : class
-        {
-            (IObjectSerializer, IObjectSerializer)[] comparisons = systemTextJsonOnly
-                ?
-                [
-                    (systemTextJsonSerializer, systemTextJsonSerializer)
-                ]
-                :
-                [
-                    (newtonsoftSerializer, newtonsoftSerializer),
-                    (newtonsoftSerializer, systemTextJsonSerializer),
-                    (systemTextJsonSerializer, newtonsoftSerializer),
-                    (systemTextJsonSerializer, systemTextJsonSerializer),
-                ];
-
-            foreach (var (serializer, deserializer) in comparisons)
-            {
-                byte[] bytes = serializer.Serialize(original);
+        protected override JsonSerializationTestCalendar Create(JObject source) => new();
+    }
 
-                WriteJson(bytes);
+    public sealed class SystemTextJsonSerializer : Serialization.SystemTextJson.CalendarSerializer<JsonSerializationTestCalendar>
+    {
+        protected override JsonSerializationTestCalendar Create(JsonElement jsonElement, StjJsonSerializerOptions options) => new();
 
-                T deserialized = deserializer.DeSerialize<T>(bytes);
+        protected override void SerializeFields(Utf8JsonWriter writer, JsonSerializationTestCalendar calendar, StjJsonSerializerOptions options)
+        {
+            writer.WriteNumber("CustomProperty", calendar.CustomProperty);
+        }
 
-                if (asserter != null)
-                {
-                    asserter(deserialized, original);
-                }
-                else
-                {
-                    deserialized.Should().Be(original);
-                }
-            }
+        protected override void DeserializeFields(JsonSerializationTestCalendar calendar, JsonElement jsonElement, StjJsonSerializerOptions options)
+        {
+            calendar.CustomProperty = jsonElement.GetProperty("CustomProperty").GetInt32();
         }
 
-        private static void WriteJson(byte[] bytes)
+        public override string CalendarTypeName => "TestCalendar";
+    }
+}
+
+public class JsonSerializationTestTrigger : SimpleTriggerImpl
+{
+    public int CustomProperty { get; set; }
+
+    public sealed class SystemTextJsonSerializer : TriggerSerializer<JsonSerializationTestTrigger>
```

Comment:
I'm not happy with the amount of code needed to support a "simple" custom trigger like in this case. However, there might not be many cases where a custom trigger is required to have serialization support so left the code as is.

Some ideas I had for improvements:
* Change the CreateScheduleBuilder to Create which returns IMutableTrigger. The TriggerConverter would then use the StaticScheduleBuilder. This way, there would be no need for the added DeserializeFields method as all custom fields could be initialized in the create method.
* Change the serializers for built-in triggers to be public and extendable (somehow...). This would make it easier to create serializers for triggers inherited from the built-in triggers. This might also be the most likely case how custom triggers are implemented.

Reply:
Yes, should be quite rare case and should probably be left for the custom trigger implementer.